cmd/deploy/deploy.go

```diff
@@ -659,101 +569,43 @@ func (dc *DeployCommand) deployDivert(ctx context.Context, opts *Options) error
 	return nil
 }
 
-func (dc *DeployCommand) cleanUp(ctx context.Context) {
-	oktetoLog.Debugf("removing temporal kubeconfig file '%s'", dc.TempKubeconfigFile)
-	if err := os.Remove(dc.TempKubeconfigFile); err != nil {
-		oktetoLog.Infof("could not remove temporal kubeconfig file: %s", err)
-	}
+func (dc *DeployCommand) deployEndpoints(ctx context.Context, opts *Options) error {
+	oktetoLog.SetStage("Endpoints configuration")
+	defer oktetoLog.SetStage("")
 
-	oktetoLog.Debugf("stopping local server...")
-	if err := dc.Proxy.Shutdown(ctx); err != nil {
-		oktetoLog.Infof("could not stop local server: %s", err)
+	c, _, err := dc.K8sClientProvider.Provide(okteto.Context().Cfg)
+	if err != nil {
+		return err
 	}
-}
-
-func newProtocolTransport(clusterConfig *rest.Config, disableHTTP2 bool) (http.RoundTripper, error) {
-	copiedConfig := &rest.Config{}
-	*copiedConfig = *clusterConfig
 
-	if disableHTTP2 {
-		// According to https://pkg.go.dev/k8s.io/client-go/rest#TLSClientConfig, this is the way to disable HTTP/2
-		copiedConfig.TLSClientConfig.NextProtos = []string{"http/1.1"}
+	iClient, err := ingresses.GetClient(ctx, c)
```

Comment:
I think we should inject this to the `DeployCommand` in the form of a provider like we do with the k8sClient, this will allow for easier testing

Reply:
🤔  as the `ingresses.GetClient()` depends on the `c` witch is set by the deployCommand params, this should have the K8sClientProvider proerty we set at DeployCommand
is this correct @ifbyol  ?

Reply:
Checking the K8sClientProvider, I've seen that it has a `GetIngressClient` method. Can we use that?

Reply:
Not needed, you can do something like:

```
type DeployCommand struct {
	...
        // This is what you would add
        getIngressesClient func(context.Context, kubernetes.Interface) (*ingresses.Client, error)
}
```

and then, you would do something like:

```
    iclient, err := dc.getIngressesClient(ctx, c)
```

In that way, when you create the command in the "production" code, you would do something like

```
    dc := &deployCommmand {
        getIngressesClient: ingresses.GetClient,
    }
```

and in the test code you could do something like

```
    dc := &deployCommmand {
        getIngressesClient: myFakeFunctionToCreateAClient,
    }
```